geode-core/src/test/java/org/apache/geode/management/internal/cli/converters/ClusterMemberIdNameConverterJUnitTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.internal.cli.converters;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.management.cli.ConverterHint;
+import org.apache.geode.test.junit.categories.UnitTest;
+import org.apache.geode.test.junit.rules.GfshParserRule;
+import org.apache.geode.test.junit.rules.GfshParserRule.CommandCandidate;
+
+@Category(UnitTest.class)
+public class ClusterMemberIdNameConverterJUnitTest {
+  @ClassRule
+  public static GfshParserRule parser = new GfshParserRule();
+  private static ClusterMemberIdNameConverter converter;
+
+  private static String[] allMemberNames = {"server1", "server2"};
+
+  @BeforeClass
+  public static void before() {
+    // this will let the parser use the spied converter instead of creating its own
+    converter = parser.spyConverter(ClusterMemberIdNameConverter.class);
+    when(converter.getCompletionValues())
+        .thenReturn(Arrays.stream(allMemberNames).collect(Collectors.toSet()));
+  }
+
+  @Test
+  public void testSupports() throws Exception {
+    assertThat(converter.supports(String.class, ConverterHint.ALL_MEMBER_IDNAME)).isTrue();
+  }
+
+  @Test
+  public void convert() throws Exception {
+    assertThat(converter.convertFromText("server", String.class, "")).isEqualTo("server");
+  }
+
+  @Test
+  public void complete() throws Exception {
+    CommandCandidate candidate = parser.complete("describe member --name=");
+    assertThat(candidate.size()).isEqualTo(allMemberNames.length);
+    assertThat(candidate.getFirstCandidate()).isEqualTo("describe member --name=server1");
+
+    candidate = parser.complete("describe member --name=");
+    assertThat(candidate.size()).isEqualTo(2);
+    assertThat(candidate.getFirstCandidate()).isEqualTo("describe member --name=server1");
```

Comment:
What coverage does this case add? Only difference between the two cases is the hard-coded expected value.

Reply:
a more extended test would be renaming this test to be BaseStringConverterTest, and spy all the converters you modified in this change set and try to do a parser.complete of all the affected parameters.

Reply:
Right - I had added this test initially specifically for `describe member` but then started down the path of refactoring all the other pieces.